src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -137,15 +137,24 @@ def _get_mgmt_service_client(cli_ctx,
                              api_version=None,
                              base_url_bound=True,
                              resource=None,
+                             scopes=None,
```

Comment:
>scopes=None, [](start = 29, length = 12)

please move `scopes` to the end of the parameter list for backward compatibility.

Reply:
Of course, I am happy to do that.

Let me ask a clarification question here. I put it next to `resource` because they have the same purpose. I thought that `_get_mgmt_service_client` is an internal private function for only internal purposes, and I checked all occurrence in the `azure-cli` project. Please let me know if considering this you still think it is good to move the parameter to the end of the function. Thank you.

Reply:
I think we should totally remove `resource` as it is an ADAL concept and has been replaced by `scopes`.

Reply:
I am happy to do this in this PR, however, it might be that removing all resource reference in one go would cleaner. Up to you. What do you think?

Reply:
@jiasli I was thinking a little bit about this, and my concern is that this function's parameter list is practically part of the public API since the `get_mgmt_service_client(...)` function just forwards the `kwargs` arguments. So, if anybody uses the `resource` named parameter by calling the `get_mgmt_service_client` method, it would break. And there are many occasions for this. Therefore, we might consider to deprecated this public function and create a new one which has only the `scopes` as accepted parameter. (This is of course just an idea, you are expert in this field.)

However, the `_get_mgmt_service_client` function internally forwards the `resource` to the `Profile::get_login_credentials` function. I think this is only for backward compatibility with the ADAL, so we can get easier rid of it. This would require, however, slightly bigger change because there are a few places in the `azure-cli` project which calls `get_login_credentials` with resource parameter. I might do this is in a separated PR.

Reply:
Let's put this on hold. We will see how we can deprecate `resource` in the future.